### PR TITLE
Restore CentOS 7.6

### PR DESCRIPTION
**Pull request by V-FEXrt**

### CODE REVIEW

**Comment by mmjconolly** [comment]
In https://github.com/sifive/wake/blob/master/.circleci/dockerfiles/centos-7.6
You could add something like:
```
yum install centos-release-scl
yum install devtoolset-9-gcc*
source scl_source enable devtoolset-9
```

**Comment by mmjconolly** [comment]
I guess we could try directly using
```
source /opt/rh/devtoolset-9/enable
```
instead of 
```
source scl_source enable devtoolset-9
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -80,6 +80,24 @@ jobs:
...
+    machine:
+      image: ubuntu-2004:202201-02
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - run: docker build -f /tmp/workspace/.circleci/dockerfiles/centos-7.6 -t wake-centos-7.6 .
+      - run: |
+          cp /tmp/workspace/wake_*.tar.xz . && \
+          cp -a /tmp/workspace/tests . && \
+          docker run --rm --device /dev/fuse --cap-add SYS_ADMIN --security-opt apparmor:unconfined --mount type=bind,source=$PWD,target=/build wake-centos-7.6 rpmbuild -ta --define "_rpmdir /build" wake_*.tar.xz && \
```
Comment (mmjconolly) [inline .circleci/config.yml:93]:
```
docker run --rm --device /dev/fuse --cap-add SYS_ADMIN --security-opt apparmor:unconfined --mount type=bind,source=$PWD,target=/build wake-centos-7.6 $some_command
```
I was thinking that the `source` would need to be part of $some_command
When `docker build` is running I think it runs each command in a fresh shell and doesn't preserve the state, so when we do `docker run` the sourced variables are missing

**Comment by mmjconolly** [comment]
What we see now is:
```
tools/job-cache/job-cache.cpp: In function 'void copy(int, int)':
tools/job-cache/job-cache.cpp:76:7: error: 'copy_file_range' was not declared in this scope; did you mean 'sync_file_range'?
```

**Comment by V-FEXrt** [comment]
That aligns with https://github.com/sifive/wake/issues/925. We will need to add some `#ifdef`s for CentOS/job cache (wild how quickly things bit rot when not protected by CI

@JakeSiFive FYI

**Comment by JakeSiFive** [comment]
I'll make a change tomorrow to do some ifdef magic.


1) We need to check for Linux 4,5 *OR* glibc version 2.27 per this: https://man7.org/linux/man-pages/man2/copy_file_range.2.html

2) For the FICLONE direct route we can just check to see if FICLONE is defined.


So we have the following cases for copy

* Apple
* Linux with no copy_file_range
* Linux with copy_file_range

Then for reflink_or_copy we have two cases

* FICLONE is defiend
* FICLONE is not defined

**Review by JakeSiFive — APPROVED** [review]

**Review by mmjconolly — APPROVED** [review]

**Comment by timsnyder-siv** [comment]
Greatly appreciated.  Thank you!

**Pull request merged**